eth2near/eth2near-block-relay-rs/src/last_slot_searcher.rs

```diff
@@ -0,0 +1,1128 @@
+            {
+                blocks.push(block)
+            }
+            slot += 1;
+        }
+
+        eth_client_contract.send_headers(&blocks, end_slot).unwrap();
+    }
+
+    #[test]
+    fn test_block_known_on_near() {
+        let mut eth_client_contract = get_client_contract(true);
+        eth_client_contract.register_submitter().unwrap();
+        let beacon_rpc_client = BeaconRPCClient::new("https://lodestar-kiln.chainsafe.io");
+        let eth1_rpc_client = Eth1RPCClient::new("https://rpc.kiln.themerge.dev");
+        let last_slot_searcher = LastSlotSearcher::new(true);
+
+        //1060486 slot without block
+        let is_block_known = last_slot_searcher.block_known_on_near(
+            1060486,
+            &beacon_rpc_client,
+            &eth_client_contract,
+        );
+        if let Ok(_) = is_block_known {
+            panic!();
+        }
+
+        let is_block_known = last_slot_searcher.block_known_on_near(
+            1099360,
+            &beacon_rpc_client,
+            &eth_client_contract,
+        );
+
+        match is_block_known {
+            Ok(is_block_known) => assert!(!is_block_known),
+            Err(_) => panic!(),
+        }
+
+        let finalized_slot = eth_client_contract
+            .get_finalized_beacon_block_slot()
+            .unwrap();
+
+        send_execution_blocks(
+            &beacon_rpc_client,
+            &mut eth_client_contract,
+            &eth1_rpc_client,
+            finalized_slot + 1,
+            finalized_slot + 1,
+        );
+
+        let is_block_known = last_slot_searcher.block_known_on_near(
+            finalized_slot + 1,
+            &beacon_rpc_client,
+            &eth_client_contract,
+        );
+        match is_block_known {
+            Ok(is_block_known) => assert!(is_block_known),
+            Err(_) => panic!(),
+        }
+    }
+
+    #[test]
+    fn test_find_left_non_error_slot() {
+        let mut eth_client_contract = get_client_contract(true);
+        eth_client_contract.register_submitter().unwrap();
+        let beacon_rpc_client = BeaconRPCClient::new("https://lodestar-kiln.chainsafe.io");
+        let eth1_rpc_client = Eth1RPCClient::new("https://rpc.kiln.themerge.dev");
+        let last_slot_searcher = LastSlotSearcher::new(true);
+
+        let (left_non_empty_slot, is_known_block) = last_slot_searcher.find_left_non_error_slot(
+            1060528,
+            1060532,
+            1,
+            &beacon_rpc_client,
+            &eth_client_contract,
+        );
+        assert_eq!(left_non_empty_slot, 1060528);
+        assert_eq!(is_known_block, false);
+
+        let (left_non_empty_slot, is_known_block) = last_slot_searcher.find_left_non_error_slot(
+            1060529,
+            1060532,
+            1,
+            &beacon_rpc_client,
+            &eth_client_contract,
+        );
+        assert_eq!(left_non_empty_slot, 1060531);
+        assert_eq!(is_known_block, false);
```

Comment:
We really need to use some constants for initial values and then apply needed amount of interval between slots

Reply:
fix: https://github.com/aurora-is-near/rainbow-bridge/pull/787/commits/e83a9b8c2fa5c7157e8813ee300566e4ca48fa14